src/compile/legend.ts

```diff
@@ -212,6 +196,24 @@ export namespace encode {
       }
     }
 
+    if (fieldDef.bin && isContinuousScale(model.scale(channel).type)) {
+      const def = {
+        scale: model.scaleName(channel),
+        field: 'value'
+      };
+      switch (channel) {
+        case OPACITY:
+          symbols.opacity = def;
+          break;
+        case SIZE:
+          symbols.size = def;
+          break;
+        case COLOR:
+          symbols[filled ? 'fill' : 'stroke'] = def;
+          break;
+      }
```

Comment:
Let's add:

```
default: 
  throw Error(`Legend for channel ${channel} not implemented`);  // put this message in log.message
```

so when someone try to add new channels, they will know suddenly that something is wrong.